src/schema/v2/notifications/index.ts

```diff
@@ -98,6 +99,68 @@ export const NotificationType = new GraphQLObjectType<any, ResolverContext>({
         return formatDate(date, dateFormat, timezone)
       },
     },
+    previewImages: {
+      type: new GraphQLList(Image.type),
+      args: {
+        size: { type: GraphQLInt },
+      },
+      resolve: async (
+        notification,
+        { size },
+        { articleLoader, artworksLoader, showsLoader, viewingRoomLoader }
+      ) => {
+        switch (notification.activity_type) {
+          case "ViewingRoomPublishedActivity": {
```

Comment:
It looks like for this activity type, as well as the article one, the `size` argument isn't being used/respected.

Should this `preview` field be a list that accepts `size`, or possibly a list that always returns all of them (no `size` argument), and the client can show just the first one (or not)? Or perhaps should this field be singular, just returning one preview image?

Reply:
Good catch! The object IDs should be sliced using the `size` parameter in every case to avoid returning more items than necessary. I've updated the resolver. 👍

Returning the list is important because the activity panel displays a list of up to 4 preview images. The `size` parameter is also needed because a notification could potentially have more than 100 object IDs.